app/models/report.rb

```diff
@@ -1434,8 +1439,27 @@ def self.report_storage_stats(report)
   end
 
   def self.report_top_uploads(report)
-    report.modes = [:table]
+    if report.filter.present?
```

Comment:
We need to find a better solution here. We can’t have  to add so much code for each report to handle filters.


I suggest you look at what we do for categories.

We might also have to rewrite sql queries using the builder: https://github.com/discourse/mini_sql

Reply:
> We need to find a better solution here. We can’t have to add so much code for each report to handle filters.

That's a good point, it seems that this can be easily extracted out into a method that can be used by any report.

> We might also have to rewrite sql queries using the builder: https://github.com/discourse/mini_sql

Awesome, I will look into that 👍